FirebaseStorage/Sources/StorageError.swift

```diff
@@ -66,7 +59,30 @@ public let StorageErrorDomain: String = "FIRStorageErrorDomain"
     if let data = (errorDictionary["data"] as? Data) {
       errorDictionary["ResponseBody"] = String(data: data, encoding: .utf8)
     }
-    return error(withCode: errorCode, infoDictionary: errorDictionary)
+    var storageError: StorageError
+    switch serverError.code {
+    case 400: storageError = StorageError.unknown(
+        message: "Unknown 400 error from backend",
+        serverError: errorDictionary
+      )
+    case 401: storageError = StorageError.unauthenticated(serverError: errorDictionary)
+    case 402: storageError = StorageError.quotaExceeded(
+        bucket: ref.path.bucket,
```

Comment:
Is passing the `bucket` needed since it's part of the `errorDictionary` (line 53)?

Reply:
I kept it to minimize the API breakage. Current users only need to add `,_)` to the `catch` to handle the breakage.

Reply:
Is the `,_)` for the `errorDictionary`?

Reply:
yes

Reply:
Thanks, resolving.